Training24/Test.c

```diff
@@ -1,14 +1,143 @@
 // ------------------------------------------------------------------------------------------------
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
 // Copyright (c) Metamation India.
+// Kalaimagal V P
 // ------------------------------------------------------------------------------------------------
-// Test.c
-// Program on main branch.
+// Test.c - The program sorts integers and searches for numbers.
 // ------------------------------------------------------------------------------------------------
 #include <stdio.h>
-#include "Header.h"
+#include <ctype.h>
+#include <stdlib.h>
+#include <conio.h>
+#include <stdbool.h>
+#include <string.h>
+#include "header.h"
 
+// ANSI escape codes for colors
+#define RESET   "\033[0m"
+#define SKYBLUE "\033[36m"
+#define GREEN   "\033[32m"
+#define YELLOW  "\033[33m"
+#define RED     "\033[31m"
+
+/// <summary>Function to clear the console screen.</summary>
+static void ClearScreen () {
+#ifdef _WIN32
+   system ("cls");
+#endif
+}
+
+/// <summary>Function to display the elements of an array.</summary>
+static void DisplayArray (int arr[], int size) {
+   printf ("| ");
+   for (int i = 0; i < size; i++) printf ("%d  ", arr[i]);
+   printf ("|\n");
+}
+
+/// <summary>Function to check if the array is sorted.</summary>
+static bool IsSorted (int arr[], int size) {
+   for (int i = 1; i < size; i++) if (arr[i - 1] > arr[i]) return false; // Not sorted
+   return true; // Sorted
+}
+
+/// <summary>Checks if the input string is a valid integer (digits only).</summary>
+static int IsValidInteger (const char* str) {
+   if (str == NULL || *str == '\n' || *str == '\0') return 0;
+   while (isspace (*str)) str++;
+   if (*str == '+' || *str == '-') str++;
+   for (; *str != '\0' && *str != '\n'; str++) if (!isdigit (*str)) return 0;
+   return 1;
+}
+
+/// <summary>Function to format and print test results.</summary>
+static void PrintTestCase (int numTests, int inputs[], int size, int searchElement) {
+   int sortedArray[50];
+   // Copy input array for sorting
+   for (int i = 0; i < size; i++) sortedArray[i] = inputs[i];
+   InsertionSort (sortedArray, size);
+   printf (YELLOW "------------------ Test Case %d -------------------\n" RESET, numTests);
+   printf ("| Input Array:            ");
+   DisplayArray (inputs, size);
+   printf ("| Output Array:           ");
+   DisplayArray (sortedArray, size);
+   printf ("| %s\n", IsSorted (sortedArray, size) ? "Insertion sort: " GREEN "PASS" RESET : "Insertion sort: " RED "FAIL" RESET);

```

Comment:
Take out the common part wherever possible. It simplifies logic.
```suggestion
printf ("|  Insertion sort : %s\n", IsSorted (sortedArray, size) ? GREEN "PASS" RESET : RED "FAIL" RESET);

```